### PR TITLE
Code Annotation

### PR DESCRIPTION
**Pull request by MrRowey**
First got at annotating the code for the Intellisense plugin.

Their are a few that are still not defined just wanted to ensure I'm on the right path

### CODE REVIEW

File: engine/Core.lua [hunk 1]
```diff
@@ -304,12 +316,16 @@ end
 
 ---
 --  Distance between two 3d points (v1,v2)
-function VDist3()
+---@param v1 number
+---@param v2 number
```
Comment (speed2CZ) [inline engine/Core.lua:320]: these are vectors, not numbers

File: engine/Core.lua [hunk 2]
```diff
@@ -304,12 +316,16 @@ end
...
+function VDist3(v1,v2)
 end
 
 ---
 --  Square of Distance between two 3d points (v1,v2)
-function VDist3Sq()
+---@param v1 number
+---@param v2 number
```
Comment (speed2CZ) [inline engine/Core.lua:327]: these are vectors, not numbers

File: lua/AI/AIBehaviors.lua [hunk 3]
```diff
@@ -55,7 +54,7 @@ function CDRRunAway(aiBrain, cdr)
                         if canTeleport then
                             IssueTeleport({cdr}, runSpot)
                         else
-                            cmd = plat:MoveToLocation(runSpot, false)
+                           local cmd = plat:MoveToLocation(runSpot, false)
```
Comment (speed2CZ) [inline lua/AI/AIBehaviors.lua:57]: this local should be removed as it's not used

File: lua/AI/AIBehaviors.lua [hunk 4]
```diff
@@ -384,6 +408,10 @@ function AirUnitRefit(self)
     end
 end
 
+---comment
+---@param unit Unit
+---@param plan integer
```
Comment (speed2CZ) [inline lua/AI/AIBehaviors.lua:413]: plan is a string with the plan name to use

File: lua/AI/AIBehaviors.lua [hunk 5]
```diff
@@ -384,6 +408,10 @@ function AirUnitRefit(self)
...
+---@param data integer
```
Comment (speed2CZ) [inline lua/AI/AIBehaviors.lua:414]: data is a table with platoon data

File: engine/Core.lua (hunk 1 above)
Comment (MrRowey) [reply engine/Core.lua:320]: Hmmm okay Vector is not a Type I'm able to select @Garanas is this something that needs adding then ?

**Comment by MrRowey** [comment]
@speed2CZ What Parameter is base then as could not fine where that is set

File: lua/AI/AIBehaviors.lua [hunk 6]
```diff
@@ -608,15 +632,9 @@ CommanderOverrideCheck = function(self)
...
--- Args:
--- platoon - the platoon
--- Description:
--- Finds the experiemental unit in the platoon (assumes platoons are only experimentals)
--- Returns:
--- experimental or nil
--------------------------------------------------------
+---Finds the experiemental unit in the platoon (assumes platoons are only experimentals)
+---@param platoon Platoon
+---@return boolean                # experimental or nil
```
Comment (hahn-kev) [inline lua/AI/AIBehaviors.lua:637]: you marked this as returning a boolean, when it looks like it actually returns a unit or nil. You can indicate that like this `@return Unit|nil` that way the person calling this function knows they should check for nil before using the value.

File: engine/Core.lua (hunk 1 above)
Comment (hahn-kev) [reply engine/Core.lua:320]: If you're not sure you can always create an alias which we can change later. So like at the top of Core you could add `---@alias Vector unknown` then any place a vector is used you can name it that, then later on we can update the alias to be the correct type.

File: engine/Core.lua [hunk 7]
```diff
@@ -161,7 +163,14 @@ end
...
 --  Create a point vector(px,py,pz, vx,vy,vz)
-function PointVector()
+---@param px number
+---@param py number
+---@param pz number
+---
+---@param vx number
+---@param vy number
+---@param vz number
+function PointVector(px,py,pz, vx,vy,vz)
```
Comment (hahn-kev) [inline engine/Core.lua:173]: make sure to specify a return type for this function and any other functions.

**Review by hahn-kev — COMMENTED** [review]
Looks like a good start, there's some functions where the comment is just `comment` but I'm assuming those are ones you haven't gotten to yet. Don't forget to specify a return type on your core functions, vector is tricky because it's a type that hasn't been defined yet, but I left a comment about that.

**Comment by 4z0t** [comment]
You must add `---@return` too for better readability and understanding

**Comment by MrRowey** [comment]
> You must add `---@return` too for better readability and understanding

from what jip told me if you do the --- for the auto complete and it dose not give a return then its not needed

File: engine/Core.lua [hunk 8]
```diff
@@ -159,9 +163,14 @@ end
 function OrientFromDir(vector)
 end
 
----
---  Create a point vector(px,py,pz, vx,vy,vz)
-function PointVector()
+---Create a point vector(px,py,pz, vx,vy,vz)
+---@param px Vector
+---@param py Vector
+---@param pz Vector
```
Comment (speed2CZ) [inline engine/Core.lua:169]: p - x y z are elements of one vector, it's not 3 vectors

File: engine/Core.lua [hunk 9]
```diff
@@ -329,12 +350,17 @@ end
...
+---@param y Vector
+---@param z Vector
+function Vector(x,y,z)
 end
 
 ---
 --  Create a vector (x,y)
-function Vector2()
+---@param x Vector
+---@param y Vector
```
Comment (speed2CZ) [inline engine/Core.lua:362]: numbers not vectors

**Pull request closed without merging**